src/olmo_core/io.py

```diff
@@ -574,15 +575,24 @@ def new_func(*args, **kwargs):
 def _http_file_size(url: str) -> int:
     response = requests.head(url, allow_redirects=True)
     content_length = response.headers.get("content-length")
-    assert content_length
+    assert (
+        content_length is not None
+    ), f"No content-length header found for {url}. Headers: {dict(response.headers)}"
     return int(content_length)
 
 
-@retriable()
+@retriable(
+    retry_condition=lambda exc: (
+        isinstance(exc, requests.exceptions.HTTPError)
+        and exc.response is not None
+        and exc.response.status_code == 502
```

Comment:
502 errors also happen a lot with `r2`, so we just retry after a couple of seconds